django/contrib/auth/tokens.py

```diff
@@ -47,41 +55,33 @@ def check_token(self, user, token):
             return False
 
         # Check that the timestamp/uid has not been tampered with
-        if not constant_time_compare(self._make_token_with_timestamp(user, ts), token):
-            # RemovedInDjango40Warning: when the deprecation ends, replace
-            # with:
-            #   return False
-            if not constant_time_compare(
-                self._make_token_with_timestamp(user, ts, legacy=True),
-                token,
-            ):
-                return False
-
-        # RemovedInDjango40Warning: convert days to seconds and round to
-        # midnight (server time) for pre-Django 3.1 tokens.
-        now = self._now()
-        if legacy_token:
-            ts *= 24 * 60 * 60
-            ts += int((now - datetime.combine(now.date(), time.min)).total_seconds())
-        # Check the timestamp is within limit.
-        if (self._num_seconds(now) - ts) > settings.PASSWORD_RESET_TIMEOUT:
+        attempts = [
+            constant_time_compare(self._make_token_with_timestamp(user, ts, secret), token)
+            for secret in self._get_secrets()
+        ]
+
+        if not constant_time_any(attempts):
+            return False
+
+        # Check the timestamp is within limit. Timestamps are rounded to
+        # midnight (server time) providing a resolution of only 1 day. If a
+        # link is generated 5 minutes before midnight and used 6 minutes later,
+        # that counts as 1 day. Therefore, PASSWORD_RESET_TIMEOUT_DAYS = 1 means
+        # "at least 1 day, could be up to 2."
+        if (self._num_days(self._today()) - ts) > settings.PASSWORD_RESET_TIMEOUT_DAYS:
```

Comment:
This looks like a bad-merge. The code has been changed since the first patch so we'll need to adjust.